### PR TITLE
Change Job Run human state names

### LINKED ISSUE #1316: Change Job Run human state name
The current human state name for a Job Run appends "(with preassembly errors)" if there were errors during preassembly (either a preassembly or a discovery job). They are labeled as "preassembly" errors to distinguish them from errors that might occur during accessioning.

Thus, human labels might include:
* Preassembly complete (with preassembly errors)
* Discovery report complete (with preassembly errors)
* Accessioning complete (with preassembly errors)

Is this the correct human names? If not, what should be?

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1316

# Why was this change made? 🤔
See ticket

# How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (e.g. changes what is written to shared file systems or how it uses APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit, Andrew

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: app/models/job_run.rb [hunk 1]
```diff
@@ -36,7 +36,6 @@ class JobRun < ApplicationRecord
 
     event :accessioning_completed do
       transition preassembly_complete: :accessioning_complete
-      transition preassembly_complete_with_errors: :accessioning_complete
```
Comment (justinlittman) [inline app/models/job_run.rb:39]: This should have previously been removed.

**Comment by andrewjbtw** [comment]
Testing revealed one more state to account for: when a job fails in Preassembly and does not proceed to accessioning.

Example: https://sul-preassembly-qa.stanford.edu/job_runs/34

This job only has one druid and that druid failed before accessioning. The status is "Running (with Preassembly errors)" but it should be "Job complete (with Preassembly errors)" because there is no more work for the job to do.

**Comment by andrewjbtw** [comment]
That's the only issue I found in my testing in terms of the state names in the UI. I did see some changes to make in the Preassembly emails.

I tested:

- discovery report error
- Preassembly error, no accessioning
- Preassembly error, at least one druid finishes accessioning
- Preassembly error, at least one druid finishes accessioning, one druid stalls in accessioning
- At least one druid finishes accessioning, one accessioning error
- Success, no errors

**Comment by andrewjbtw** [comment]
I've ticketed the email changes separately.

**Comment by justinlittman** [comment]
Fixed the problem reported by @andrewjbtw so ready for review.

File: config/locales/en.yml [hunk 2]
```diff
@@ -37,11 +37,10 @@ en:
           states:
             running: 'Running'
             discovery_report_complete: 'Discovery report completed'
-            preassembly_complete: 'Preassembly completed'
+            preassembly_complete: 'Running'
```
Comment (edsu) [inline config/locales/en.yml:40]: Is it worth leaving a comment here about why `preassembly_complete` shares the label `running`?

File: spec/models/job_run_spec.rb [hunk 3]
```diff
@@ -101,4 +101,89 @@
...
+        context 'when preassembly errors' do
+          let(:error_message) { 'Drat' }
+
+          it 'returns name' do
+            expect(name).to eq('Job completed (with preassembly errors)')
+          end
+        end
+      end
+    end
+  end
```
Comment (edsu) [inline spec/models/job_run_spec.rb:188]: 🎉 for thorough tests of this!

**Review by edsu — APPROVED** [review]

File: config/locales/en.yml (hunk 2 above)
Comment (justinlittman) [reply config/locales/en.yml:42]: Good idea. Added.

**Pull request merged**